client/core/core.go

```diff
@@ -705,48 +682,76 @@ func (c *Core) ConnectWallet(assetID uint32) error {
 	return err
 }
 
-// PreRegister creates a connection to the specified DEX and fetches the
+func (c *Core) isDEXRegistered(url string) bool {
+	c.connMtx.RLock()
+	_, found := c.conns[url]
+	c.connMtx.RUnlock()
+	return found
+}
+
+// PreRegister creates a connection to the specified DEX Server and fetches the
 // registration fee. The connection is left open and stored temporarily while
 // registration is completed.
 func (c *Core) PreRegister(form *PreRegisterForm) (uint64, error) {
-	c.connMtx.RLock()
-	_, found := c.conns[form.URL]
-	c.connMtx.RUnlock()
-	if found {
+	if c.isDEXRegistered(form.URL) {
 		return 0, fmt.Errorf("already registered at %s", form.URL)
 	}
 
-	dc, err := c.connectDEX(&db.AccountInfo{
-		URL:  form.URL,
-		Cert: []byte(form.Cert),
-	})
-	if err != nil {
-		return 0, err
+	// Don't (re-)connect if there is a cached connection to this DEX server.
+	dexConn := c.connCache.conn
+	if dexConn == nil || dexConn.acct.url != form.URL {
+		dc, err := c.connectDEX(&db.AccountInfo{
+			URL:  form.URL,
+			Cert: []byte(form.Cert),
+		})
+		if err != nil {
+			return 0, err
+		}
+		dexConn = dc
+
+		// New connection established, close previously cached connection.
+		c.closeTemporaryConn()
```

Comment:
I'm starting to think the whole cached connection thing is a mistake. It seems like an unnecessary micro-optimization. What do you think about getting rid of it altogether? Maybe rename `Preregister` to `GetFee` or something more descriptive of what it actually does, and then just close and dump the connection after fee retrieval.

Reply:
I like that name change.

Reply:
I'd actually renamed `PreRegister` to `CheckDEXFees` and `Register` to `RegisterDEX` to be more descriptive of what they do. I'll push that update in a bit.

Reply:
Re the connection caching, I think it's fine rn since we'd typically expect a user to check fees before registering, we shouldn't connect to the same server multiple times _if_ we can avoid it.
If it was not already implemented, we could just ignore it but since we already have the micro-optimization, any need to get rid of it?
Another use of the connection caching feature is for when a register attempt fails and user tries again.
In any case, we could always establish a connection w/o stress each time `PreRegister/GetFee` and `Register` is called, so no solid direction preference from me.
Maybe someone has a stronger opinion?

Reply:
> I'd actually renamed `PreRegister` to `CheckDEXFees` and `Register` to `RegisterDEX` to be more descriptive of what they do.

We're not going to start adding `DEX` into the name of every method. It's a DEX client package. The DEX part is implied.

Reply:
> we shouldn't connect to the same server multiple times if we can avoid it.

I think there's a trade off with code complexity here. The complexity around connection caching already created one bug here, and it certainly doesn't help readability. Now you're looking to add an additional 4~0 loc and 2 type methods to deal with it. I don't think it's worth it, personally.

Reply:
Right. Off it goes then.